src/named_threads.cc

```diff
@@ -0,0 +1,42 @@
+#include "named_threads.h"
+
+#include <cassert>
+
+#include <fmt/format.h>
+
+namespace celerity::detail::named_threads {
+
+thread_type tt_device_submitter(const uint32_t n) {
+	assert(n < thread_type_step);
+	return thread_type(static_cast<uint32_t>(thread_type::first_device_submitter) + n); // NOLINT(clang-analyzer-optin.core.EnumCastOutOfRange)
+}
+thread_type tt_host_queue(const uint32_t n) {
+	assert(n < thread_type_step);
+	return thread_type(static_cast<uint32_t>(thread_type::first_host_queue) + n); // NOLINT(clang-analyzer-optin.core.EnumCastOutOfRange)
+}
+thread_type tt_test(const uint32_t n) {
+	assert(n < thread_type_step);
+	return thread_type(static_cast<uint32_t>(thread_type::first_test) + n); // NOLINT(clang-analyzer-optin.core.EnumCastOutOfRange)
+}
+
+std::string thread_type_to_string(const thread_type t_type) {
+	switch(t_type) {
+	case thread_type::application: return "cy-application";
+	case thread_type::scheduler: return "cy-scheduler";
+	case thread_type::executor: return "cy-executor";
+	case thread_type::alloc: return "cy-alloc";
+	default: break;
+	}
+	if(t_type >= thread_type::first_device_submitter && t_type < thread_type::first_host_queue) { //
+		return fmt::format("cy-be-sub-{}", static_cast<uint32_t>(t_type) - static_cast<uint32_t>(thread_type::first_device_submitter));
```

Comment:
How about `cy-submit-{}`? "be" == backend is not even part of the enum variant anymore.

Reply:
Went with `dev-sub` since you make a good point about trying to match the enum.